sonic-ledd/scripts/ledd

```diff
@@ -56,50 +57,67 @@ class DaemonLedd(daemon_base.DaemonBase):
-            self.log_warning("sel.select() did not return swsscommon.Select.OBJECT")
-            return 2
+            self.subscribeDbTable("APPL_DB", swsscommon.APP_PORT_TABLE_NAME, namespace)
+    
+    def connectDB(self, dbname, namespace):
+        db = daemon_base.db_connect(dbname, namespace=namespace)
+        return db
+    
+    def subscribeDbTable(self, dbname, tblname, namespace):
+        db = self.connectDB(dbname, namespace)
+        self.tables[namespace] = swsscommon.SubscriberStateTable(db, tblname)
+        self.sel.addSelectable(self.tables[namespace])
+
+    def isFrontPanelPort(self, port_name):
+        return not port_name.startswith((backplane_prefix(), inband_prefix(), recirc_prefix()))
+
+    def updatePortLedColor(self, port_name, port_status):
+        self.led_control.port_link_state_change(port_name, port_status)
+
+    def getEventNamespace(self, selectObj):
+        # Get the corresponding namespace from redisselect db connector object
+        return selectObj.getDbConnector().getNamespace()
+    
+    def processPortTableEvent(self, selectableObj):
+        ''' Process (if any) event from the PORT table in the Application DB '''
 
         # Get the redisselect object from selectable object
         redisSelectObj = swsscommon.CastSelectableToRedisSelectObj(selectableObj)
+        namespace = self.getEventNamespace(redisSelectObj)
 
-        # Get the corresponding namespace from redisselect db connector object
-        namespace = redisSelectObj.getDbConnector().getNamespace()
-
-        (key, op, fvp) = self.sst[namespace].pop()
+        (key, op, fvp) = self.tables[namespace].pop()
         if fvp:
             # TODO: Once these flag entries have been removed from the DB,
             # we can remove this check
             if key in ["PortConfigDone", "PortInitDone"]:
-                return 3
+                return
 
             fvp_dict = dict(fvp)
 
-            if op == "SET" and "oper_status" in fvp_dict:
-                if not key.startswith((backplane_prefix(), inband_prefix(), recirc_prefix())):
-                    self.led_control.port_link_state_change(key, fvp_dict["oper_status"])
-        else:
-            return 4
+            if op == "SET" and "oper_status" in fvp_dict and self.isFrontPanelPort(key):
+                self.updatePortLedColor(key, fvp_dict["oper_status"])
+
+
+
+    # Run daemon
+    def run(self):
+        # Use timeout to prevent ignoring the signals we want to handle
+        # in signal_handler() (e.g. SIGTERM for graceful shutdown)
+        (state, selectableObj) = self.sel.select(SELECT_TIMEOUT)
+
+        if state == swsscommon.Select.TIMEOUT:
+            # NOOP - Nothing to process here
+            return 0
+
+        if state != swsscommon.Select.OBJECT:
+            self.log_warning("sel.select() did not return swsscommon.Select.OBJECT - May be socket closed???")
+            return -1 ## Fail here so that the daemon can be restarted
```

Comment:
Consider return code = (128+non-zero-code) per all other daemons code.

Reply:
@assrinivasan do you know the reason for 128+ ?